File: 완전탐색/소수 찾기.py
from itertools import permutations

def is_prime(num):
    if num <= 1:
        return False
    for i in range(2,num):
        if num%i==0:
            return False
    return True

def solution(numbers):
    nums = set()
    answer = []
    for i in range(len(numbers)):
        comb = list(permutations(numbers,i+1))
        for num in comb:
            nums.add(int(''.join(num)))
    for num in nums:
        if is_prime(num):
            answer.append(num)

    return len(answer)

File: 고득점kit/완전탐색/소수찾기.js
function solution() {

function isPrime(num) {
    if (num < 2) {
      return false;
    }
  
    for (let i = 2; i <= Math.sqrt(num); i++) {
      if (num % i === 0) {
        return false;
      }
    }
  
    return true;
  }
  
  function findPrimePermutations(nums) {
    const result = [];
  
    function backtrack(nums, current) {
      if (current.length > 0 && isPrime(Number(current.join('')))) {
        result.push(Number(current.join('')));
      }
  
      for (let i = 0; i < nums.length; i++) {
        if (i > 0 && nums[i] === nums[i - 1]) {
          continue; // 중복된 숫자는 건너뜁니다.
        }
        current.push(nums[i]);
        backtrack(nums, current);
        current.pop();
      }
    }
  
    nums.sort((a, b) => a - b); // 숫자 배열을 정렬합니다.
    backtrack(nums, []);
  
    return result;
  }
  
  // 예제 #1
  const nums1 = [1, 7];
  const primePermutations1 = findPrimePermutations(nums1);
  console.log(`예제 #1: ${primePermutations1}`);
  
  // 예제 #2
  const nums2 = [0, 1, 1];
  const primePermutations2 = findPrimePermutations(nums2);
  console.log(`예제 #2: ${primePermutations2}`);

}